### ISSUE TITLE
Ability to have canonical root domain without redirect.

### ISSUE THREAD

**Issue by Daltz333** [readthedocs/readthedocs.org#8250]
## Details

* Read the Docs project URL: frc-docs.readthedocs.io/
* Build URL (if applicable): na
* Read the Docs username (if applicable): daltz333

As of the deployment of #3231, the "Canonical" option under "Domains" no longer just sets the canonical meta data in non-root webpages, but also redirects the other pages to the "Root".

This comment from #3231 helps detail our use-case
> It's definitely the case that canonical=True was poorly named or poorly applied when we originally wrote the implementation. Ideally canonical=True would function the way you're describing. From our current perspective, canonical=True only affects how the resolver determines the domain for the project and pieces like search and auth on the commercial side -- its effectively primary=True, not canonical=True.

We have a "primary" domain that we would prefer all search engines and SEO to use. This is https://docs.wpilib.org/. However, we also have a secondary domain https://frcdocs.wpi.edu/ that we direct users toward when the domain "https://docs.wpilib.org/" is blocked. It's often just domain blocked by over-aggressive school internet (this is our target demographic as we write docs for a high-school robotics competition).

**Comment by stsewd** [comment]
Hmm, I think if you uncheck the canonical option from the other domain the page will be served from that domain (not sure), if that works you can manually set the canonical meta tags with this option https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_baseurl

**Comment by Daltz333** [comment]
I have unchecked the option, but it still redirects. It may take a bit for things to resolve (guessing). If it works, I'll follow that root to set the canonical meta tags.

**Comment by stsewd** [comment]
Looks like now it redirects to the rtd.io domain :D

`````````````````````````````````````
$ curl -I https://docs.wpilib.org                                                        
HTTP/2 302 
date: Wed, 09 Jun 2021 15:59:52 GMT
content-type: text/html; charset=utf-8
content-length: 0
location: https://frc-docs.readthedocs.io/en/stable/
```

**Comment by Daltz333** [comment]
Redirecting to the RTD domain isn't the correct behavior either. We want all the domains to be equally accessible, but one domain "preferred"

**Comment by Daltz333** [comment]
Hm. It looks like multiple domains is completely broken. This should probably get changed to "bug". It's either all domains redirect to readthedocs, or redirect to the canonical root.

**Comment by stsewd** [comment]
> It's either all domains redirect to readthedocs, or redirect to the canonical root.

Yeah, that's the current behavior, it was implemented in that way.

**Comment by Daltz333** [comment]
What is the point in adding multiple domains besides redirecting?

**Comment by stsewd** [comment]
I agree this shouldn't be the expected result, that's why I labeled this as `design decision`, I think this is mostly related to our resolver code always pointing the one domain.

**Comment by Daltz333** [comment]
Gotcha. Thanks.
